cli/src/pcluster/config/validators.py

```diff
@@ -797,6 +797,43 @@ def s3_bucket_validator(param_key, param_value, pcluster_config):
     return errors, warnings
 
 
+def s3_bucket_region_validator(param_key, param_value, pcluster_config):
+    """Validate S3 bucket is in the same region with the cloudformation stack."""
+    errors = []
+    warnings = []
+    s3_client = boto3.client("s3")
+    try:
+        bucket_region = s3_client.get_bucket_location(Bucket=param_value).get("LocationConstraint")
+        # Buckets in Region us-east-1 have a LocationConstraint of null
+        # Example output from get_bucket_location for us-east-1:
+        #   {'ResponseMetadata': {...}, 'LocationConstraint': None}
+        if bucket_region is None:
+            bucket_region = "us-east-1"
+        if bucket_region != pcluster_config.region:
+            errors.append("cluster_resource_bucket must be in the same region of the cluster.")
+    except ClientError as client_error:
+        if client_error.response.get("Error").get("Code") == "NoSuchBucket":
+            errors.append(
+                "The S3 bucket '{0}' does not appear to exist: '{1}'".format(
+                    param_value, client_error.response.get("Error").get("Message")
+                )
+            )
+        elif client_error.response.get("Error").get("Code") == "AccessDenied":
+            errors.append(
+                "You do not have access to the S3 bucket '{0}': '{1}'".format(
+                    param_value, client_error.response.get("Error").get("Message")
+                )
+            )
+        else:
+            errors.append(
+                "Unexpected error when calling get_bucket_location on S3 bucket '{0}': '{1}'".format(
+                    param_value, client_error.response.get("Error").get("Message")
+                )
+            )
```

Comment:
You can probably call `_process_generic_s3_bucket_error` to avoid code duplication.
We will need to fix the error message `Unexpected error when calling get_bucket_location on S3 bucket` in `_process_generic_s3_bucket_error`, because it should be a generic error.

nit: we can also fix similar code duplications in `fsx_lustre_auto_import_validator
` below

Reply:
Done! Aslo fixed that for `fsx_lustre_auto_import_validator`